packages/contact-exchange/src/services/OOBParser.ts

```diff
@@ -0,0 +1,43 @@
+import { OutOfBandInvitation } from './DIDCommOOBInvitation';
+
+export interface ParsedOOB {
+  invitation: OutOfBandInvitation;
+  encodedPart: string;
+}
+
+export function parseOOBInvitation(url: string): ParsedOOB | null {
+  const urlPattern = /^(https:\/\/mediator\.rootsid\.cloud\?_oob=)(.*)$/;
+  const match = url.match(urlPattern);
+  if (!match) {
+    return null;
+  }
+
+  const encodedPart = match[2];
+  try {
+    const decodedData = atob(encodedPart); // Decoding base64
+    const invitation: OutOfBandInvitation = JSON.parse(decodedData);
+
+    // Validate the invitation object
+    if (!invitation['@id'] || !invitation['@type']) {
```

Comment:
This validation fails when I try in the tests:

let validEncodedUrl = 'https://mediator.rootsid.cloud?_oob=eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9vdXQtb2YtYmFuZC8yLjAvaW52aXRhdGlvbiIsImlkIjoiMDczODZhZTUtMjYzYi00ZjgwLWE2M2ItZmI5OTE1ODIzM2IyIiwiZnJvbSI6ImRpZDpwZWVyOjIuRXo2TFNtczU1NVloRnRobjFXVjhjaURCcFptODZoSzl0cDgzV29qSlVteFBHazFoWi5WejZNa21kQmpNeUI0VFM1VWJiUXc1NHN6bTh5dk1NZjFmdEdWMnNRVllBeGFlV2hFLlNleUpwWkNJNkltNWxkeTFwWkNJc0luUWlPaUprYlNJc0luTWlPaUpvZEhSd2N6b3ZMMjFsWkdsaGRHOXlMbkp2YjNSemFXUXVZMnh2ZFdRaUxDSmhJanBiSW1ScFpHTnZiVzB2ZGpJaVhYMCIsImJvZHkiOnsiZ29hbF9jb2RlIjoicmVxdWVzdC1tZWRpYXRlIiwiZ29hbCI6IlJlcXVlc3RNZWRpYXRlIiwibGFiZWwiOiJNZWRpYXRvciIsImFjY2VwdCI6WyJkaWRjb21tL3YyIl19fQ'